### PR TITLE
Use cpplint

### LINKED ISSUE #46: Use cpplint
[cpplint](https://github.com/cpplint/cpplint) is a code checker for C/C++. It can be used in the CI to check the code style, what do you think @FrancescAlted?

### PR DESCRIPTION
**Pull request by aleixalcacer**

### CODE REVIEW

File: caterva/caterva.h [hunk 1]
```diff
@@ -9,32 +9,27 @@
...
  * @brief Caterva header file.
  *
  * This file contains Caterva public API and the structures needed to use it.
  * @author Blosc Development team <blosc@blosc.org>
  */
 
-#ifndef CATERVA_HEADER_FILE
-#define CATERVA_HEADER_FILE
+#ifndef CATERVA_CATERVA_H_
+#define CATERVA_CATERVA_H_
```
Comment (FrancescAlted) [inline caterva/caterva.h:20]: Too much redundancy in `CATERVA_CATERVA` prefix?

File: caterva/caterva.h [hunk 2]
```diff
@@ -9,32 +9,27 @@
...
-#define CATERVA_VERSION_MINOR    4   /* for minor interface/format changes  */
-#define CATERVA_VERSION_RELEASE  1-dev   /* for tweaks, bug-fixes, or
- * development */
-
-#define CATERVA_VERSION_STRING   "0.4.1-dev"  /* string version. Sync with
- * above! */
-#define CATERVA_VERSION_DATE     "2020-07-28"    /* date version */
+#define CATERVA_VERSION_MAJOR 0         /* for major interface/format changes  */
+#define CATERVA_VERSION_MINOR 4         /* for minor interface/format changes  */
+#define CATERVA_VERSION_RELEASE 1 - dev /* for tweaks, bug-fixes, or development */
```
Comment (FrancescAlted) [inline caterva/caterva.h]: Are the additional blanks in `1 - dev` intended?

File: caterva/caterva.h [hunk 3]
```diff
@@ -108,23 +121,19 @@ typedef struct {
...
+static const caterva_config_t CATERVA_CONFIG_DEFAULTS = {.alloc = malloc,
+                                                         .free = free,
+                                                         .compcodec = BLOSC_ZSTD,
+                                                         .complevel = 5,
+                                                         .usedict = 0,
+                                                         .nthreads = 1,
+                                                         .filters = {0, 0, 0, 0, 0, BLOSC_SHUFFLE},
+                                                         .filtersmeta = {0, 0, 0, 0, 0, 0},
+                                                         .prefilter = NULL,
+                                                         .pparams = NULL};
```
Comment (FrancescAlted) [inline caterva/caterva.h:136]: I don't see why the new format for `caterva_config_t` is better than the previous one.

**Review by FrancescAlted — CHANGES_REQUESTED** [review]
I like this one!  Please address the suggestions that I made and let's merge as soon as possible.

File: caterva/caterva.h (hunk 1 above)
Comment (aleixalcacer) [reply caterva/caterva.h:20]: It is determined by cpplint. If it is changed to `CATERVA_H_`, then:
```
caterva/caterva.h:19:  #ifndef header guard has wrong style, please use: CATERVA_CATERVA_H_  [build/header_guard] [5]
caterva/caterva.h:494:  #endif line should be "#endif  // CATERVA_CATERVA_H_"  [build/header_guard] [5]
```

File: caterva/caterva.h (hunk 2 above)
Comment (aleixalcacer) [reply caterva/caterva.h]: Fixed :)

File: caterva/caterva.h (hunk 3 above)
Comment (aleixalcacer) [reply caterva/caterva.h:136]: To adapt Caterva to cpplint requirements, `clang-format` is used:

```
clang-format -style=file -i caterva/* 
```

And it changes it automatically. What do you think about using this workflow?

**Pull request merged**